### DISCUSSION TITLE
The purpose of getPatternIndex and getId

### DISCUSSION THREAD

**Discussion by samedit66** [bonede/tree-sitter-ng#14] [Q&A]
Hello! I'm trying to use a query to extract nodes from the tree. My code:
```java
        TSQuery all_declarators = new TSQuery(_language, "(variable_declarator) @decls");
        TSQueryCursor cursor = new TSQueryCursor();
        cursor.exec(all_declarators, node);
        TSQueryMatch match = new TSQueryMatch();
        while (cursor.nextMatch(match)) {

        }
```
I found out that match has methods called getPatternIndex and getId, but what are they used for? I though getId might return index of the node of interest but it seems not... I tried getPatternIndex but stuck with the same result

**Comment by bonede** [comment]
Hi.

The getPatternIndex() method should return the pattern index in the query. The getId() method returns the id of the match(index in the match list for now).
